app/models/form526_submission.rb

```diff
@@ -566,4 +566,47 @@ def poll_form526_pdf
   def cleanup
     EVSS::DisabilityCompensationForm::SubmitForm526Cleanup.perform_async(id)
   end
+
+  def account
+    # first, check for an ICN on the UserAccount associated to the submission, return it if found
+    account = user_account
+    return account if account.present? && account.icn.present?
+
+    # next, check past submissions for different UserAccounts that might have ICNs
+    past_submissions = get_past_submissions
+    account = find_user_account_with_icn(past_submissions, 'past submissions')
+    return account if account.present? && account.icn.present?
+
+    # next, check for any historical UserAccounts for that user which might have an ICN
+    user_verifications = get_user_verifications
+    account = find_user_account_with_icn(user_verifications, 'user verifications')
+    return account if account.present? && account.icn.present?
+
+    # failing all the above, default to an Account lookup
+    Account.lookup_by_user_uuid(user_uuid)
+  end
+
+  def find_user_account_with_icn(records, record_type)
+    records.pluck(:user_account_id).uniq.each do |user_account_id|
+      user_account = UserAccount.find(user_account_id)
+      if user_account&.icn.present?
+        Rails.logger.info("ICN not found on submission #{id}, " \
+                          "using ICN for user account #{user_account_id} instead (based on #{record_type})")
+        return user_account
+      end
+    end
+  end
+
+  def get_past_submissions
+    Form526Submission.where(user_uuid: user_uuid).where.not(user_account_id: user_account_id)
```

Comment:
This query can be shortened using keyword argument syntax, i.e., `Form526Submission.where(user_uuid:).where.not(user_account_id:)`